javav2/usecases/Creating_Spring_Redshift_ Rest/src/test/java/RedshiftRestTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+import com.aws.rest.RetrieveItems;
+import com.aws.rest.SendMessage;
+import com.aws.rest.WorkItem;
+import com.aws.rest.WriteExcel;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestMethodOrder;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Properties;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@TestInstance(TestInstance.Lifecycle.PER_METHOD)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class RedshiftRestTest {
+    private static String id="";
+    private static String email="";
+
+    @BeforeAll
+    public static void setUp() {
+
+        try (InputStream input = RedshiftRestTest.class.getClassLoader().getResourceAsStream("config.properties")) {
+
+            Properties prop = new Properties();
+            if (input == null) {
+                System.out.println("Sorry, unable to find config.properties");
+                return;
+            }
+
+            prop.load(input);
+            id = prop.getProperty("id");
+            email = prop.getProperty("email");
+
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    @Test
+    @Order(1)
+    public void checkValues() {
+        assertNotNull(id);
+        assertNotNull(email);
+        System.out.println("Test 1 passed");
+    }
+
+    @Test
+    @Order(2)
+    public void invokeArchive() {
+        RetrieveItems ri = new RetrieveItems();
+        List<WorkItem> myList = ri.getData(1);
+        assertNotNull(myList);
```

Comment:
While asserting the list is not null (empty) here, you should also check that it's the size you expect it to be. If it should always be a size 1 list, the assert the list's size is 1.

Reply:
This is a dynamic value - it depends on the records in the table. Therefore we cannot specify a fixed size.